### PR TITLE
[FEAT] connect: add modulus operator and withColumns support

### PR DESCRIPTION
**Pull request by andrewgazelka**
- Add `%` operator and `sum` function to unresolved functions
- Implement withColumns transformation
- Add test coverage for group by with modulus operation

### CODE REVIEW

File: tests/connect/test_group_by.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # Add a column that will have repeated values for grouping
+    df = df.withColumn("group", col("id") % 3)
+
+    # Group by the new column and sum the ids in each group
+    df_grouped = df.groupBy("group").sum("id")
+
+    # Convert to pandas to verify the sums
+    df_grouped_pandas = df_grouped.toPandas()
+
+    print(df_grouped_pandas)
```
Comment (jaychia) [inline tests/connect/test_group_by.py:19]: rm?

File: src/daft-connect/src/translation/logical_plan/with_columns.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            };
+
+            to_daft_expr(&expression)
+        })
+        .try_collect()?;
+
+    let plan = plan.with_columns(daft_exprs)?;
+
+    Ok(plan)
+}
```
Comment (jaychia) [inline src/daft-connect/src/translation/logical_plan/with_columns.rs:32]: Little confused -- is this PR adding `with_columns` or adding groupby?

**Review by jaychia — APPROVED** [review]
LGTM, probably change the PR messaging though to reflect that we're adding `sum` and `with_columns`. Looks like groupby itself already works.

File: src/daft-connect/src/translation/logical_plan/with_columns.rs (hunk 2 above)
Comment (andrewgazelka) [reply src/daft-connect/src/translation/logical_plan/with_columns.rs:32]: we dicussed this; changing title of this PR.

**Pull request merged**